src/support.js

```diff
@@ -602,24 +602,29 @@ function makeBigInt(low, high, unsigned) {
 
 /** @param {Array=} args */
 function dynCall(sig, ptr, args) {
-  if (args && args.length) {
-#if ASSERTIONS
-    // j (64-bit integer) must be passed in as two numbers [low 32, high 32].
-    assert(args.length === sig.substring(1).replace(/j/g, '--').length);
-#endif
-#if ASSERTIONS
-    assert(('dynCall_' + sig) in Module, 'bad function pointer type - no table for sig \'' + sig + '\'');
-#endif
-    return Module['dynCall_' + sig].apply(null, [ptr].concat(args));
-  } else {
-#if ASSERTIONS
-    assert(sig.length == 1);
-#endif
+#if !WASM_BIGINT
+  // Without WASM_BIGINT support we cannot directly call function with i64 as
+  // part of thier signature, so we rely the dynCall functions generated by
+  // wasm-emscripten-finalize
+  if (sig.indexOf('j') != -1) {
 #if ASSERTIONS
     assert(('dynCall_' + sig) in Module, 'bad function pointer type - no table for sig \'' + sig + '\'');
+    if (args && args.length) {
```

Comment:
```suggestion
    if (args) {
```
an empty array is truthy in JS (but not in python)

Reply:
This code is pre-existing, plus it seems like we need both check here.. we want to check for args being defined *and* having a length.

If we remove the first check we will get `Cannot read property 'length' of undefined`.. if we remove the second check as you suggest when the empty array will go down the wrong path (since as you say its truthy).

Reply:
I'm going to land this as is for the above reasons.. if you think i'm wrong we can fix after.

Reply:
Oh, right, I was mistaken about `args.length`, lgtm.